bookdown/03-optimization-feature-selection.Rmd

```diff
@@ -109,4 +109,171 @@ Work in progress.
+
+### The `FSelectInstance` Classes {#fs-wrapper-optimization}
+
+The following sub-section examines the feature selection on the `r ref("mlr_tasks_sonar", text = "Pima")` data set which is used to predict whether or not a patient has diabetes. 
+
+```{r 03-optimization-feature-selection-006}
+task = tsk("pima")
+print(task)
+```
+We use the classification tree from `r cran_pkg("rpart")`.
+
+```{r 03-optimization-feature-selection-007}
+learner = lrn("classif.rpart")
+```
+
+Next, we need to specify how to evaluate the performance of the feature subsets.
+For this, we need to choose a `r ref("Resampling", text = "resampling strategy")` and a `r ref("Measure", text = "performance measure")`.
+
+```{r 03-optimization-feature-selection-008}
+hout = rsmp("holdout")
+measure = msr("classif.ce")
+```
+
+Finally, one has to choose the available budget for the feature selection.
+This is done by selecting one of the available `r ref("Terminator", text = "Terminators")`:
+
+* Terminate after a given time (`r ref("TerminatorClockTime")`)
+* Terminate after a given amount of iterations (`r ref("TerminatorEvals")`)
+* Terminate after a specific performance is reached (`r ref("TerminatorPerfReached")`)
+* Terminate when feature selection does not improve (`r ref("TerminatorStagnation")`)
+* A combination of the above in an *ALL* or *ANY* fashion (`r ref("TerminatorCombo")`)
+
+For this short introduction, we specify a budget of 20 evaluations and then put everything together into a `r ref("FSelectInstanceSingleCrit")`:
+
+```{r 03-optimization-feature-selection-009}
+library("mlr3fselect")
+
+evals20 = trm("evals", n_evals = 20)
+
+instance = FSelectInstanceSingleCrit$new(
+  task = task,
+  learner = learner,
+  resampling = hout,
+  measure = measure,
+  terminator = evals20
+)
+instance
+```
+
+To start the feature selecton, we still need to select a feature selection algorithm which are defined via the `r ref("FSelector")` class.
+
+### The `FSelector` Class
+
+The following algorithms are currently implemented in `r mlr_pkg("mlr3fselect")`:
+
+* Random Search (`r ref("FSelectorRandomSearch")`)
+* Exhaustive Search (`r ref("FSelectorExhaustiveSearch")`)
+* Sequential Search (`r ref("FSelectorSequential")`)
+* Recursive Feature Elimination (`r ref("FSelectorRFE")`)
+* Design Points (`r ref("FSelectorDesignPoints")`)
+
+In this example, we will use a simple random search.
+
+```{r 03-optimization-feature-selection-010}
+fselector = fs("random_search")
+```
+
+### Triggering the Tuning {#wrapper-selection-triggering}
+
+To start the feature selection, we simply pass the `r ref("FSelectInstanceSingleCrit")` to the `$optimize()` method of the initialized `r ref("FSelector")`. The algortihm proceeds as follows:
+
+1. The `r ref("FSelector")` proposes at least one feature subset  (the `r ref("FSelector")` may propose multiple subsets to improve parallelization, which can be controlled via the setting `batch_size`).
```

Comment:
this reads slightly confusingly to me, how about: "The `r ref("FSelector")` proposes at least one feature subset and may propose multiple subsets to improve parallelization, which can be controlled via the setting `batch_size`."